Add com.github.dubstar.design manifest

dubstar-04 opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

hfiguiere commented on wood.dan.design.json:
```diff
@@ -0,0 +1,50 @@
+{
+    "app-id" : "wood.dan.design",
```
`dan.wood` doesn't exist. I don't even think `wood` is a valid TLD.

hfiguiere commented on wood.dan.design.json:
```diff
@@ -0,0 +1,50 @@
+    ],
+    "modules" : [
+        {
+      "name": "blueprint",
```
this needs a cleaniup rule to remove it completely as it is a build tool.

dubstar-04 replied:
Do you know of any examples for this?

dubstar-04 replied:
just add the path to the cleanup section?

hfiguiere commented on wood.dan.design.json:
```diff
@@ -0,0 +1,50 @@
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/dubstar-04/Design.git"
```
this needs a spec revision (tag or commit)

dubstar-04 commented:
If I have correctly fixed the issues I would like this added to flathub-beta

hfiguiere commented on com.github.dubstar.design.json:
```diff
@@ -0,0 +1,54 @@
+{
+    "app-id" : "com.github.dubstar.design",
```
to should be `io.github`

https://docs.flatpak.org/en/latest/conventions.html#application-ids

hfiguiere replied:
Also the org / user name being what it is, the acceptable app-id should be `io.github.dubstar_04.design`

hfiguiere commented on com.github.dubstar.design.json:
```diff
@@ -0,0 +1,54 @@
+                {
+                    "type" : "git",
+                    "path" : "https://github.com/dubstar-04/Design.git",
+                    "tag": "/v/Design/43-alpha1",
```
this not a valid tag.

`git tag -l` will tell you

dubstar-04 commented:
> The metainfo validation fails on the screenshot urls. ![image](https://user-images.githubusercontent.com/1136710/215180369-1abcd1e7-5e21-4c6b-8ba7-83f525f98321.png)
> 
> The links are valid. Is there an issue with github urls?

This is now resolved

hfiguiere commented:
these URL point to the page, not the raw file.

hfiguiere commented:
bot, build io.github.dubstar_04.design

hfiguiere commented on io.github.dubstar_04.design.json:
```diff
@@ -0,0 +1,54 @@
+            "sources" : [
+                {
+                    "type" : "git",
+                    "path" : "https://github.com/dubstar-04/Design.git",
```
```suggestion
                    "url" : "https://github.com/dubstar-04/Design.git",
```

dubstar-04 commented:
Is there anything I need to do to progress this?

barthalion commented:
bot, build io.github.dubstar_04.design

barthalion commented:
This fails on `Check for 128x128 icon` now. Typically, you need to cross-check 3 things: whether there's an icon in somewhere `/app/share/`, the content of metainfo file and the desktop file. The metainfo reads the `Icon` field in the desktop file, and then looks for the file itself. It has to match the app ID.

hfiguiere commented:
The launchable is wrong 
https://github.com/dubstar-04/Design/blob/43-alpha1/data/io.github.dubstar_04.design.metainfo.xml

I'm pretty sure it is not set properly at build time since this is not a `.in` file.

dubstar-04 commented:
> The launchable is wrong https://github.com/dubstar-04/Design/blob/43-alpha1/data/io.github.dubstar_04.design.metainfo.xml
> 
> I'm pretty sure it is not set properly at build time since this is not a `.in` file.

The launchable has to match the desktop filename? with '.in' included

hfiguiere commented:
I'm pretty it's unchanged at installation

![image](https://user-images.githubusercontent.com/114441/216054929-fb306871-5070-4754-b671-24d20b2be597.png)

it it's wrong.

hfiguiere commented:
actually it duplicates the one at the top. which is probably why it fails.

hfiguiere commented:
on a side note moving tags is a bad idea. and a tag that that the same name as a branch, also a bad idea.

dubstar-04 commented:
Are we able to try building again?

dubstar-04 commented:
bot, build io.github.dubstar_04.design

hfiguiere commented:
> @hfiguiere hopefully thats fixed now. Thank you for all your help.

you removed the wrong one. So it still fails.

dubstar-04 commented:
> > @hfiguiere hopefully thats fixed now. Thank you for all your help.
> 
> you removed the wrong one. So it still fails.

I don't understand. I renamed the desktop file and removed the duplicate launchable tag. The error is regarding a missing icon?

hfiguiere commented:
You removed this:

<img width="1087" alt="image" src="https://user-images.githubusercontent.com/114441/216125416-472ca6d0-d5b1-4ef4-8b2b-6f0aa403c071.png">


This one is correct. The other one is not.

dubstar-04 commented:
bot, build io.github.dubstar_04.design

dubstar-04 commented:
@hfiguiere thank you for all your help. Really appreciated.

dubstar-04 commented:
Installs and works as expected! 

![image](https://user-images.githubusercontent.com/1136710/216128169-4b62ed8c-b7e0-49f6-b61b-7fb51c98cb87.png)

barthalion commented:
/merge
